src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.cs

```diff
@@ -137,5 +138,373 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
 C.M
 Program.Main");
         }
+
+        [Fact]
+        public void MultipleInitializers_SingleFile()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C1
+{
+    [ModuleInitializer]
+    internal static void M1() => Console.Write(1);
+
+    internal class C2
+    {
+        [ModuleInitializer]
+        internal static void M2() => Console.Write(2);
+    }
+
+    [ModuleInitializer]
+    internal static void M3() => Console.Write(3);
+}
+
+class Program 
+{
+    static void Main() => Console.Write(4);
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+
+            CompileAndVerify(
+                source,
+                parseOptions: s_parseOptions,
+                options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    var rootModuleType = (TypeSymbol)module.GlobalNamespace.GetMember("<Module>");
+                    var staticConstructor = (PEMethodSymbol)rootModuleType.GetMember(".cctor");
+
+                    Assert.NotNull(staticConstructor);
+                    Assert.Equal(MethodKind.StaticConstructor, staticConstructor.MethodKind);
+
+                    var expectedFlags =
+                        MethodAttributes.Private
+                        | MethodAttributes.Static
+                        | MethodAttributes.SpecialName
+                        | MethodAttributes.RTSpecialName
+                        | MethodAttributes.HideBySig;
+
+                    Assert.Equal(expectedFlags, staticConstructor.Flags);
+                },
```

Comment:
Should this be repeated? I'm not repeating it in any of the new tests in the diagnostics PR.

Reply:
(This includes the ` options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),` line too)

Reply:
It seems fine to extract the validator method and reuse it.

Reply:
> Should this be repeated? I'm not repeating it in any of the new tests in the diagnostics PR.

Are you referring to assert for flags? I think it is fine to test that once, but it is your call

---
In reply to: [420305655](https://github.com/dotnet/roslyn/pull/43964#discussion_r420305655) [](ancestors = 420305655)

Reply:
Not just the flags but the symbol validation itself, since verification is being run. Though I see now that it only runs when `ExecutionConditionUtil.IsMonoOrCoreClr`.

Reply:
> Not just the flags but the symbol validation itself, since verification is being run. Though I see now that it only runs when ExecutionConditionUtil.IsMonoOrCoreClr.

There is no conditional logic in how we emit metadata for the initializer. I would say testing symbol properties once is sufficient, but, if you'd like, feel free to test in more scenarios. 

---
In reply to: [420323727](https://github.com/dotnet/roslyn/pull/43964#discussion_r420323727) [](ancestors = 420323727)